cashu/mint/ledger.py

```diff
@@ -643,20 +651,17 @@ async def mint(
         logger.trace("called mint")
         amounts = [b.amount for b in B_s]
         amount = sum(amounts)
-        async with self.db.connect() as conn:
-            logger.trace("attempting lock table invoice")
-            await conn.execute(lock_table(self.db, "invoices"))
-            logger.trace("locked table invoice")
-            # check if lightning invoice was paid
-            if settings.lightning:
-                if not hash:
-                    raise Exception("no hash provided.")
-                try:
-                    logger.trace("checking lightning invoice")
-                    paid = await self._check_lightning_invoice(amount, hash, conn)
-                    logger.trace(f"invoice paid: {paid}")
-                except Exception as e:
-                    raise e
+
+        if settings.lightning:
+            if not hash:
+                raise Exception("no hash provided.")
+            self.locks[hash] = (
+                self.locks.get(hash) or mp.Lock()
+            )  # create a new lock if it doesn't exist

```

Comment:
Could make `self.locks` a `defaultdict` so it creates for you?

Reply:
Seems like `defaultdict` doesn't accept classes by default (hehe) and would need some additional boilerplate: https://stackoverflow.com/questions/32932494/defaultdict-of-a-class-get-index-key-of-current-called-instance-inside-a-class